code/controllers/stonedmc/subsystem/title.dm

```diff
@@ -0,0 +1,69 @@
+SUBSYSTEM_DEF(title)
+	name = "Title Screen"
+	flags = SS_NO_FIRE
+	init_order = INIT_ORDER_TITLE
+
+	var/file_path
+	var/icon/icon
+	var/icon/previous_icon
+	var/turf/closed/wall/indestructible/splashscreen/splash_turf
+
+/datum/controller/subsystem/title/Initialize()
+	if(file_path && icon)
+		return
+
+	if(fexists("data/previous_title.dat"))
+		var/previous_path = file2text("data/previous_title.dat")
+		if(istext(previous_path))
+			previous_icon = new(previous_icon)
+	fdel("data/previous_title.dat")
+
+	var/list/provisional_title_screens = flist("[global.config.directory]/title_screens/images/")//wtf, this also picks non-image shits
+	var/list/title_screens = list()
+	var/use_rare_screens = prob(1)
+
+	SSmapping.HACK_LoadMapConfig()
+	for(var/S in provisional_title_screens)
+		var/list/L = splittext(S,"+")
+		if((L.len == 1 && L[1] != "blank.png")|| (L.len > 1 && ((use_rare_screens && lowertext(L[1]) == "rare") || (lowertext(L[1]) == lowertext(SSmapping.config.map_name)))))
+			title_screens += S
+
+	if(length(title_screens))
+		file_path = "[global.config.directory]/title_screens/images/[pick(title_screens)]"
+
+	if(!file_path)
+		file_path = "icons/misc/title.dmi"
+
+	ASSERT(fexists(file_path))
+
+	icon = new(fcopy_rsc(file_path))
+
+	if(splash_turf)
+		splash_turf.icon = icon
+
+	return ..()
+
+/datum/controller/subsystem/title/vv_edit_var(var_name, var_value)
+	. = ..()
+	if(.)
```

Comment:
see: orignal tg code of title.dm

Reply:
Just because /tg/ has bad code doesn't mean we have to have it too.
Currently, if `.` is false, it returns null, same as if you flipped the statement there